aw_test.go

```diff
@@ -350,6 +350,46 @@ var _ = Describe("Peer", func() {
 			Expect(receivedData).To(Equal(rightContent))
 		})
 
+		It("should expire old pending syncs", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			opts := defaultOptions(logger)
+			opts.MaxPendingSyncs = 10
+
+			peer1 := newPeerAndListen(ctx, opts)
+			peer2 := newPeerAndListen(ctx, opts)
+			peers := []*aw.Peer{peer1, peer2}
+
+			connectAllPeers(peers)
+
+			go peer1.Run(ctx)
+			go peer2.Run(ctx)
+
+			linkAllPeers(peers)
+
+			successID := []byte("success")
+			peer2.ContentResolver.InsertContent(successID, []byte("content"))
+
+			for i := 0; i < int(opts.MaxPendingSyncs); i++ {
+				i := i
+				go func() {
+					contentID := []byte(fmt.Sprintf("%v", i))
+					syncCtx, syncCancel := context.WithTimeout(ctx, 100*time.Millisecond)
+					peer1.Sync(syncCtx, contentID, nil)
+					syncCancel()
+				}()
+			}
```

Comment:
I wonder if we should wait for all these pending syncs finished before trying a new one. i.e. adding a `WaitGroup`

Reply:
Fixed in 532e566.